File: system/src/main/java/com/prodigal/system/controller/SpaceAnalyzeController.java
package com.prodigal.system.controller;

import com.prodigal.system.common.BaseResult;
import com.prodigal.system.common.ResultUtils;
import com.prodigal.system.exception.BizStatus;
import com.prodigal.system.exception.ThrowUtils;
import com.prodigal.system.model.dto.space.analyze.*;
import com.prodigal.system.model.entity.Space;
import com.prodigal.system.model.entity.User;
import com.prodigal.system.model.vo.space.analyze.*;
import com.prodigal.system.service.SpaceAnalyzeService;
import com.prodigal.system.service.UserService;
import org.springframework.web.bind.annotation.*;

import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import java.util.List;

/**
 * @program: prodigal-picture
 * @author: Lang
 * @description: 空间图库分析
 **/
@RestController
@RequestMapping("/space/analyze")
public class SpaceAnalyzeController {
    @Resource
    private UserService userService;
    @Resource
    private SpaceAnalyzeService spaceAnalyzeService;

    /**
     * 空间使用情况分析
     * @param spaceUsageAnalyzeDto 分析条件
     * @param request 请求
     * @return
     */
    @PostMapping("/usage")
    public BaseResult<SpaceUsageAnalyzeVO> analyzeSpaceUsage(@RequestBody  SpaceUsageAnalyzeDTO spaceUsageAnalyzeDto, HttpServletRequest request) {
        ThrowUtils.throwIf(spaceUsageAnalyzeDto == null, BizStatus.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        SpaceUsageAnalyzeVO spaceUsageAnalyzeVO = spaceAnalyzeService.analyzeSpaceUsage(spaceUsageAnalyzeDto, loginUser);
        return ResultUtils.success(spaceUsageAnalyzeVO);
    }
    /**
     * 空间图片分类情况分析
     * @param spaceCategoryAnalyzeDto 分析条件
     * @param request 请求
     * @return
     */
    @PostMapping("/category")
    public BaseResult<List<SpaceCategoryAnalyzeVO>> analyzeSpaceCategory(@RequestBody SpaceCategoryAnalyzeDTO spaceCategoryAnalyzeDto, HttpServletRequest request) {
        ThrowUtils.throwIf(spaceCategoryAnalyzeDto == null, BizStatus.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        List<SpaceCategoryAnalyzeVO> spaceCategoryAnalyzeVOList = spaceAnalyzeService.analyzeSpaceCategory(spaceCategoryAnalyzeDto, loginUser);
        return ResultUtils.success(spaceCategoryAnalyzeVOList);
    }

    /**
     * 空间图片标签情况分析
     * @param spaceTagAnalyzeDto 分析条件
     * @param request
     * @return
     */
    @PostMapping("/tag")
    public BaseResult<List<SpaceTagAnalyzeVO>> analyzeSpaceTag(@RequestBody SpaceTagAnalyzeDTO spaceTagAnalyzeDto, HttpServletRequest request) {
        ThrowUtils.throwIf(spaceTagAnalyzeDto == null, BizStatus.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        List<SpaceTagAnalyzeVO> spaceTagAnalyzeVOList = spaceAnalyzeService.analyzeSpaceTag(spaceTagAnalyzeDto, loginUser);
        return ResultUtils.success(spaceTagAnalyzeVOList);
    }

    /**
     * 空间图片大小情况分析
     * @param spaceSizeAnalyzeDto 分析条件
     * @param request
     * @return
     */
    @PostMapping("/size")
    public BaseResult<List<SpaceSizeAnalyzeVO>> analyzeSpaceSize(@RequestBody SpaceSizeAnalyzeDTO spaceSizeAnalyzeDto, HttpServletRequest request) {
        ThrowUtils.throwIf(spaceSizeAnalyzeDto == null, BizStatus.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        List<SpaceSizeAnalyzeVO> spaceSizeAnalyzeVOList = spaceAnalyzeService.analyzeSpaceSize(spaceSizeAnalyzeDto, loginUser);
        return ResultUtils.success(spaceSizeAnalyzeVOList);
    }

    /**
     * 用户上传行为分析
     * @param spaceUserAnalyzeDto 分析条件
     * @param request
     * @return
     */
    @PostMapping("/user")
    public BaseResult<List<SpaceUserAnalyzeVO>> analyzeSpaceUser(@RequestBody SpaceUserAnalyzeDTO spaceUserAnalyzeDto, HttpServletRequest request) {
        ThrowUtils.throwIf(spaceUserAnalyzeDto == null, BizStatus.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        List<SpaceUserAnalyzeVO> spaceUserAnalyzeVOList = spaceAnalyzeService.analyzeSpaceUser(spaceUserAnalyzeDto, loginUser);
        return ResultUtils.success(spaceUserAnalyzeVOList);
    }

    /**
     * 空间使用排名分析
     * @param spaceRankAnalyzeDto 分析条件
     * @param request
     * @return
     */
    @PostMapping("/rank")
    public BaseResult<List<Space>> analyzeSpaceRank(@RequestBody SpaceRankAnalyzeDTO spaceRankAnalyzeDto, HttpServletRequest request) {
        ThrowUtils.throwIf(spaceRankAnalyzeDto == null, BizStatus.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        List<Space> spaceList = spaceAnalyzeService.analyzeSpaceRank(spaceRankAnalyzeDto, loginUser);
        return ResultUtils.success(spaceList);
    }
}

File: prodigal-picture-ui/src/api/spaceAnalyzeController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request.ts'

/** analyzeSpaceCategory POST /api/space/analyze/category */
export async function analyzeSpaceCategoryUsingPost(
  body: API.SpaceCategoryAnalyzeDto,
  options?: { [key: string]: any }
) {
  return request<API.BaseResultListSpaceCategoryAnalyzeVO_>('/api/space/analyze/category', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** analyzeSpaceRank POST /api/space/analyze/rank */
export async function analyzeSpaceRankUsingPost(
  body: API.SpaceRankAnalyzeDto,
  options?: { [key: string]: any }
) {
  return request<API.BaseResultListSpace_>('/api/space/analyze/rank', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** analyzeSpaceSize POST /api/space/analyze/size */
export async function analyzeSpaceSizeUsingPost(
  body: API.SpaceSizeAnalyzeDto,
  options?: { [key: string]: any }
) {
  return request<API.BaseResultListSpaceSizeAnalyzeVO_>('/api/space/analyze/size', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** analyzeSpaceTag POST /api/space/analyze/tag */
export async function analyzeSpaceTagUsingPost(
  body: API.SpaceTagAnalyzeDto,
  options?: { [key: string]: any }
) {
  return request<API.BaseResultListSpaceTagAnalyzeVO_>('/api/space/analyze/tag', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** analyzeSpaceUsage POST /api/space/analyze/usage */
export async function analyzeSpaceUsageUsingPost(
  body: API.SpaceUsageAnalyzeDto,
  options?: { [key: string]: any }
) {
  return request<API.BaseResultSpaceUsageAnalyzeVO_>('/api/space/analyze/usage', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** analyzeSpaceUser POST /api/space/analyze/user */
export async function analyzeSpaceUserUsingPost(
  body: API.SpaceUserAnalyzeDto,
  options?: { [key: string]: any }
) {
  return request<API.BaseResultListSpaceUserAnalyzeVO_>('/api/space/analyze/user', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}
